pkg/validators/kubeletconfigvalidator.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validators
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/version"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/k8stopologyawareschedwg/deployer/pkg/clientutil"
+	"github.com/k8stopologyawareschedwg/deployer/pkg/validator"
+	"github.com/openshift-kni/numaresources-operator/controllers"
+	"github.com/openshift-kni/numaresources-operator/pkg/kubeletconfig"
+
+	nropv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+	"github.com/openshift-kni/numaresources-operator/pkg/machineconfigpools"
+	mcov1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+type KubeletConfigDataGatherer interface {
+	CollecData() (*kubeletConfigValidatorData, error)
+}
+
+type k8KubeletConfigDataGatherer struct {
+	conn    k8sConnection
```

Comment:
this seems a bit un-idiomatic

Reply:
Maybe but it could help us with validator testing. Splitting the concerns about getting the information to validate and validate it will allow us to ease the testing creating  "dataGatherer" for testing with the kubeletconfigs we need to setup the testing.

Reply:
thing is the pair of `clientset` and `context` doesn't really form a "connection". We do have primitives to fake/use empty `context`s (`context.TODO(), which we overuse already) and to mock the kube clients. I believe that the testing side has indeed big merits, but we can address them by splitting the fetch of the objects from their validation, can't we?

Reply:
Yes we can, but that's the main point for this interface,  to split information fetch from validation and to be able to create different ways to fetch that info from different sources, hiding the source from the validation process itself.

Surely the name it is not the proper one, I just though this were the "things" I was using to connect to k8s so I give them that name ... 
But I do not understand the  fake context/client thing ... I mean this concrete implementation needs client/context to get data but maybe another one with "hardcoded" `kubeletConfigValidatorData` for unit testing will not need it, or even use this one with the fake client/context in e2e testing... I am surely missing something sorry, but I do not get it

Reply:
Let's start from the `k8sconnection` object. This is un-idiomatic because isn't really a connection. Context reuse is hard, so I'd not put it inside a struct - let's just use a new one each time and pass to nested function as needed -when needed-.
So now we can just have a clientset inside the `k8KubeletConfigDataGatherer`.
Then let's talk `k8KubeletConfigDataGatherer`. I'll have another pass but I'm not convinced we need this abstraction just yet.